### ISSUE TITLE
Topbar - Creating torus doesn't use Cursor position when triggered from another editor

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#4841]
# Topbar Result
`bpy.ops.mesh.primitive_torus_add(major_radius=1, minor_radius=0.25, abso_major_rad=1.25, abso_minor_rad=0.75)`

# Operator in 3D View Result (correct)
`bpy.ops.mesh.primitive_torus_add(align='WORLD', location=(1.7214, 2.95051, -0.768177), rotation=(0, 0, 0))`

Seems something is off, as running the operator from another editor in Blender works fine. Maybe this operator has changes we did that  slipped through with something that overrides the Blender method.

**Comment by Blenux** [comment]
Yeah I did tried to look into this once before, but could never figure it out, weird one since other primitives are fine.

**Comment by Blenux** [comment]
well never mind, found it lol

**Comment by Blenux** [comment]
The cause is this commit [here](https://github.com/Bforartists/Bforartists/commit/1fe427b461611fb4e3b91aa53c7a7c3e3faf9e27#diff-87f3c0b577b82014b60a475b1e588e6353ed5088c6ee06f14c72048073c1825c),  on the object_utils.py file, if that is removed, the torus will be added to where the 3dcursor is.

though I dont know what the commit on that file was for exactly?

**Comment by ReinerBforartists** [comment]
When this code comes with a merge then this change comes from Blender. But i see BFA comments.

**Comment by Draise14** [comment]
Could you.. clarify? What is removed to make it work? The view alignment workaround? Or only the comments?

**Comment by Blenux** [comment]
oh will have to recheck this one, forgot what I changed.

**Comment by Blenux** [comment]
@Draise14 if one removes theses, 

![image](https://github.com/user-attachments/assets/73be19c7-815b-40a1-ae76-d6c2b4568def)

then the torus will be added where the 3d cursor is, 

![image](https://github.com/user-attachments/assets/cef9e267-65c6-4181-b0c7-507ef658aeb6)
